### PR TITLE
Fix 4535 - support file uploads with filenames containing single quote

### LINKED ISSUE #4535: File upload unusual behaviour on a ' character
### Describe the bug

A file or folder with the single quote ' character in the path name will exhibit unusual behaviour on the upload.

For example if uploading to /data/te'st/file.txt

The following two folders will be created. All square brackets[] added here to emphasize individual folders, they do not show up in the test.

**Empty folder:**
/[data]/[te'st]/

**Folder with empty file.txt**
/[data]/[te\st && base64 -d - >]/[data]/[te\\st]/file.txt

### Fabric8 Kubernetes Client version

5.12.4

### Steps to reproduce

```
String path = "/data/te'st/file.txt";
            client
                    .pods()
                    .withName(podName)
                    .file(path)
                    .upload(inputStream);
```

### Expected behavior

**Empty folder:**
/data/te'st/

**Folder with empty file.txt**
/data/te\st && base64 -d - >/data/te\\st/file.txt

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.24

### Environment

Linux

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

Looks to be the area that's causing the issue is here:
https://github.com/fabric8io/kubernetes-client/blob/0f997aade423fe2612250096cf87015d040d7099/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/uploadable/PodUpload.java#L139

Specifically inside the shellQuote function:
```
  public static String shellQuote(String value) {
    return "'" + value.replace("'", "'\\\\''") + "'";
  }
```

### PR DESCRIPTION
**Pull request by BrianDevC**
## Description

Fix #4535

Single quotes in folder or file names were causing corruption of the shell command. The fix changes the ' character to '"'"'
For example a folder called te'st would now get shellQuoted to 'te'"'"'st'
1. ' ends the first quote,
2. " starts second quote,
3. ' is valid character in double quotes
4. " ends the second quote
5. ' starts the third quote
6. The end result when fed into the shell, is as we desired: te'st

https://github.com/fabric8io/kubernetes-client/issues/4535

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [x] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by BrianDevC** [comment]
> Looks good. It's also possible to just change \\\\ to be \\. Can you add/modify a test to PodIT.uploadFile? - there's one there already with a bunch of special chars, but not a single quote.

I tried that but didn't get the results I expected.
Now trying it again it works just fine. Perhaps my sanity is leaving me. I'll re-patch to that. Adding a new assertion to the test you suggested.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 ### 6.2.0 (2022-10-20)
 
 #### Bugs
+* Fix #4535: The shell command string will now have single quotes sanitized
```
Comment (shawkins) [inline CHANGELOG.md:18]: Please move this to the 6.3 section above.  

Unfortunately we don't yet have an automated way of generating these entries as part of the release process.

**Comment by BrianDevC** [comment]
The code smells look to be A, not caused by my changes and B, fairly simple to fix. (Change throwable to exception.)

I'm happy to go ahead and commit that assuming tests still pass to keep it happy but am I alright to do so under this issue? Don't know your policy on lumping technically different issues together under the same number. :)

**Comment by shawkins** [comment]
> I'm happy to go ahead and commit that assuming tests still pass to keep it happy but am I alright to do so under this issue? 

Sonar isn't perfect so it can find stuff that is out of scope or false positives.  It won't hold up this pr to have code check failing, @manusa will merge it regardless.  

If you'd like to address these, I don't have a problem with that.

> Don't know your policy on lumping technically different issues together under the same number. :)

As long as they are this straight-forward, it's fine to have them under the same PR.  As separate commits would be best.

**Review by manusa — APPROVED** [review]

**Pull request merged**